libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -207,3 +223,193 @@ def verify_storage_set(self, storage_set_id: str = None,
+            flat_list.extend(elements)
+        date_cmd = "{}-{}-{}".format(flat_list[1], flat_list[0].split("=")[1], flat_list[3])
+        self.log.info("Date from command: %s", date_cmd)
+        my_date = datetime.strptime(get_response_date, "%Y-%m-%d")
+        date_resp = "{}-{}-{}".format(my_date.day,
+                                            my_date.strftime("%b"), my_date.year)
+        self.log.info("Date from response: %s", date_resp)
+        if date_cmd != date_resp:
+            err_msg = "Date Verification failed."
+            self.log.error(err_msg)
+            result = False
+        else:
+            self.log.info("Date Verification successful")
+        self.log.info("Verify time in date details")
+        time_cmd = flat_list[2]
+        self.log.info("Time from get_response: %s", time_cmd)
+        self.log.info("Time from command: %s", time_cmd)
+        if time_cmd != get_response_time:
+            err_msg = err_msg + " Time details match failed"
+            self.log.error(err_msg)
+        else:
+            self.log.info("Time details match successful")
+        return result, err_msg
+
+    def verify_serial_number(self, get_response_number: int):
+        """
+        Verify serial number for certificate
+        """
+        err_msg = ""
+        result = True
+        cmd = cmds.CMD_DECRYPT_CERTIFICATE.format("serial")
+        resp = self.master.execute_cmd(cmd=cmd, read_lines=True)
+        resp = resp.decode() if isinstance(resp, bytes) else resp
+        self.log.info("Print resp: %s", resp)
+        output_cmd = [res.strip().split("=")[1] for res in resp]
+        output_cmd = int((''.join(output_cmd)), 16)
+        self.log.info("Print serial number from response %s ", get_response_number)
+        self.log.info("Print serial number from command %s", output_cmd)
+        if output_cmd == get_response_number:
+            self.log.info("Serial number match successful")
+        else:
+            err_msg = "Serial number match failed"
+            self.log.error(err_msg)
+            result = False
+        return result, err_msg
+
+    def verify_version_number(self, get_response_version: str):
+        """
+        Verify certificate version
+        """
+        err_msg = ""
+        result = True
+        resp = self.master.execute_cmd(
+                cmd=cmds.CMD_DECRYPT_CERTIFICATE.format("text"), read_lines=True)
+        resp = resp.decode() if isinstance(resp, bytes) else resp
+        self.log.info("Print resp: %s", resp)
+        for version_cmd in resp:
+            if "version" in version_cmd.lower():
+                version_cmd = "v" + version_cmd.strip().split()[1]
+                self.log.info("Printing version from command %s", version_cmd)
+                break
+        get_response_version = get_response_version.split(".")[1]
+        self.log.info("Printing get response version %s", get_response_version)
+        if version_cmd == get_response_version:
```

Comment:
result = version_cmd == get_response_version

Reply:
do you mean removing result = True from above and modifying this to if result = res_dct == issuer_dict; ?